src/ice/mod.rs

```diff
@@ -1038,16 +1056,64 @@ mod test {
                 progress_count: 0,
                 time: now,
                 drop_sent_packets: false,
+                nat: None,
             }
         }
 
+        pub fn with_symmetric_nat(mut self, external_ip: &str) -> Self {
+            self.nat = Some(Nat::new_symmetric(external_ip));
+            self
+        }
+
+        pub fn with_port_restricted_nat(mut self, external_ip: &str) -> Self {
+            self.nat = Some(Nat::new_port_restricted_cone(external_ip));
+            self
+        }
+
         fn add_host_candidate(&mut self, addr: &str) -> Candidate {
             self.agent
                 .add_local_candidate(host(addr, "udp"))
                 .unwrap()
                 .clone()
         }
 
+        fn add_relay_candidate(&mut self, addr: &str, local: &str) -> Candidate {
+            let local = local.parse().unwrap();
+            let addr = addr.parse().unwrap();
+
+            self.agent
+                .add_local_candidate(Candidate::relayed(addr, local, "udp").unwrap())
+                .unwrap()
+                .clone()
+        }
+
+        fn server_reflexive_candidate(&mut self, stun_server: &str, base: &str) -> Candidate {
```

Comment:
Why not `add_xxx`? here for consistency?

Reply:
Because server-reflexive candidates are usually not added to the local client, they are just sent to the remote.

Reply:
In a way, this just performs STUN. I'll add a comment to clarify.